protocols/gossipsub/src/behaviour/tests.rs

```diff
@@ -228,6 +228,32 @@ mod tests {
         peer
     }
 
+    fn disconnect_peer<D, F>(gs: &mut Gossipsub<D, F>, peer_id: &PeerId)
+    where
+        D: DataTransform + Default + Clone + Send + 'static,
+        F: TopicSubscriptionFilter + Clone + Default + Send + 'static,
+    {
+        if let Some(peer_connections) = gs.connected_peers.get(peer_id) {
+            let fake_endpoint = ConnectedPoint::Dialer {
+                address: Multiaddr::empty(),
+                role_override: Endpoint::Dialer,
+            }; // this is not relevant
+               // peer_connections.connections should never be empty.
+            let mut active_connections = peer_connections.connections.len() - 1;
+            for conn_id in peer_connections.connections.clone() {
+                let handler = gs.new_handler();
+                gs.inject_connection_closed(
+                    peer_id,
+                    &conn_id,
+                    &fake_endpoint,
+                    handler,
+                    active_connections,
+                );
+                active_connections = active_connections.saturating_sub(1);
```

Comment:
```suggestion
                active_connections = active_connections.checked_sub(1).unwrap();
```

In which case would an overflow be valid? Would the above help catch future bugs?

Reply:
we are counting the remaining connections after removal of one, so we start in `active - 1` (because we want to inform of how many are remaining after this one). This line updates the count for _the next removal_. So this means that when the last connection is removed we are already in 0 remaining connections. So basically the suggested line would always panic for the last connection

Reply:
This can probably be rewritten in a better way. I'll look into it

Reply:
https://github.com/libp2p/rust-libp2p/pull/2445/commits/2d47bd344fe5127d666752d5e37eca8adb5a3612